src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,151 @@
+package duke;
+
+import java.time.DateTimeException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DeadlineCommand;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.ErrorCommand;
+import duke.command.EventCommand;
+import duke.command.FindCommand;
+import duke.command.HelloCommand;
+import duke.command.ListCommand;
+import duke.command.SnoozeCommand;
+import duke.command.TodoCommand;
+import duke.command.UndoCommand;
+
+/**
+ * Parser object that parses all the input from user to commands understood by Duke
+ */
+public class Parser {
```

Comment:
Perhaps you can add javadoc for the public methods in this class since they seem to be non-trivial?